plugins/modules/systems_info.py

```diff
@@ -0,0 +1,97 @@
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils.sap_launchpad_systems_runner import *
+from ..module_utils.sap_id_sso import sap_sso_login
+
+from requests.exceptions import HTTPError
+
+DOCUMENTATION = r'''
+---
+module: systems_info
+
+short_description: Queries registered systems in me.sap.com
+
+version_added: 1.1.0
+
+options:
+  suser_id:
+    description:
+      - SAP S-User ID.
+    required: true
+    type: str
+  suser_password:
+    description:
+      - SAP S-User Password.
+    required: true
+    type: str
+  filter:
+    description:
+      - An ODATA filter expression to query the systems.
+    required: true
+    type: str
+author:
+    - Lab for SAP Solutions
+
+'''
+
+
+EXAMPLES = r'''
+- name: get system by SID and product
+  community.sap_launchpad.systems_info:
+    suser_id: 'SXXXXXXXX'
+    suser_password: 'password'
+    filter: "Insnr eq '12345678' and sysid eq 'H01' and ProductDescr eq 'SAP S/4HANA'"
```

Comment:
```suggestion
      suser_id: S1234
      filter:
        - Insnr:
            operator: 'eq'
            value: '123456789'
        - sysid:
            operator: 'eq'
            value: 'B01'
        - ProductDescr:
            operator: 'eq'
            value: 'SAP S/4HANA'
```

Reply:
interesting idea. while I see your point of expressing it more yaml-natively, I think this won't work:
- what are the semantic of the list? are the items joined as an AND? or an OR? how is the other one expressed then?
- so if you want to properly model ODATA filters using yaml, you'd have to basically come up with a small YAML DSL (imagine nested `and:` and `or:`). is this worth the additional layer of abstraction? I think since ODATA filters are a well known and standardized form, I'd just have the user pass it it directly a string. It's also very nice since you can just copy it from browsing around manually on https://launchpad.support.sap.com/services/odata/i7p/odata/bkey :)

Reply:
Inline filtering is good to me as well was just an idea. But may be we should reference the link in the description to help users find it and use it :-)

Reply:
suggested a description